app/main/posts/views/post-toolbar.html

```diff
@@ -6,36 +6,45 @@
 
     <filter-posts filters="filters" embed-only="false"></filter-posts>
 
-    <div class="button-group" embed-only="false" ng-switch="editMode.editing">
-        <post-share ng-switch-when="false" filters="filters" button="true"></post-share>
-        <button ng-switch-when="false" ng-if="selectedPost && hasPermission && editEnabled()" class="button button-alpha" ng-click="setEditMode()"translate>app.edit</button>
-        <button ng-switch-when="true" ng-if="hasPermission" class="button" ng-click="setEditMode()" translate>app.cancel</button>
-        <button ng-switch-when="true" ng-if="hasPermission && !savingPost.saving" ng-click="savePost()" class=" button button-alpha" translate>app.save</button>
-        <button ng-switch-when="true" class="button-alpha" disabled ng-if="savingPost.saving" translate>app.saving
-            <div class="loading">
-                <div class="line"></div>
-                <div class="line"></div>
-                <div class="line"></div>
-            </div>
-          </button>
-
+    <div class="button-group" embed-only="false">
+        <div ng-if="!editMode()">
+            <post-share filters="filters" button="true"></post-share>
+            <button ng-if="editEnabled()" class="button button-alpha" ng-click="setEditMode()" translate>app.edit</button>
+        </div>
+
+        <div ng-if="editMode()">
+            <button ng-if="hasPermission" class="button" ng-click="cancel()" translate>app.cancel</button>
```

Comment:
It kinda looks like I could bump the `hasPermission` check up a couple of levels since we don't show much of anything without it.